clients/cobol-lsp-vscode-extension/src/services/ProcessorGroupsLoader.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2024 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+import * as path from "path";
+import * as fs from "fs";
+import { SettingsUtils } from "./util/SettingsUtils";
+import { PathReporter } from "io-ts/lib/PathReporter";
+
+const PROCESSOR_GROUP_FOLDER = ".cobolplugin";
+const PROCESSOR_GROUP_PGM = "pgm_conf.json";
+const PROCESSOR_GROUP_PROC = "proc_grps.json";
+
+export type ProgramsConfig = {
+  pgms: {
+    program: string;
+    pgroup: string;
+  }[];
+};
+
+export type Preprocessor =
+  | string
+  | {
+      name: string;
+      libs?: string[];
+      [key: string]: string | string[] | undefined;
+    };
+
+export type ProcessorGroup = {
+  name: string;
+  preprocessor?: Preprocessor | Preprocessor[];
+  libs?: string[];
+  "copybook-extensions"?: string[];
+  "copybook-file-encoding"?: string;
+  "compiler-options"?: string[];
+};
+
+export const readProgramConfigFileContent = (): ProgramsConfig => {
+  const EMPTY = { pgms: [] };
+  const ws = SettingsUtils.getWorkspaceFoldersPath(true);
+  if (ws.length < 1) {
+    return EMPTY;
+  }
+  const cfgPath = path.join(ws[0], PROCESSOR_GROUP_FOLDER);
+  const pgmCfgPath = path.join(cfgPath, PROCESSOR_GROUP_PGM);
+  if (!fs.existsSync(pgmCfgPath)) {
+    return EMPTY;
+  }
+  try {
+    return JSON.parse(fs.readFileSync(pgmCfgPath).toString());
```

Comment:
Now that the io-ts has be added to the project, the incoming any type could actually be validated.